### PR TITLE
Reworked the import topology functionality

### PR DESCRIPTION
**Pull request by zelenikty**
Reworked the import topology functionality by using a POST request to the server now.

- Start and end date: 2018-05-20 to 2018-05-21
- Contributor: Thommy Zelenik, @zelenikty
- Supervisor: Karoline Saatkamp, @saatkamp, Lukas Harzenetter, @lharzenetter, Oliver Kopp, @koppor

Used POST request instead of own implementation.

- [x] Ensure that you followed http://eclipse.github.io/winery/dev/ToolChain#github---prepare-pull-request.
- [x] Branch name checked. That means, the branch name starts with `thesis/`, `fix/`, ...
- [x] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Ensure to use auto format in **all** files
- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for UI changes)

### CODE REVIEW

**Comment by genie-winery** [comment]
Can one of the admins verify this patch?

**Comment by lharzenetter** [comment]
Please resolve the confilcts first.

File: org.eclipse.winery.topologymodeler.ui/package-lock.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "@angular-devkit/build-optimizer": {
       "version": "0.0.42",
       "resolved": "https://registry.npmjs.org/@angular-devkit/build-optimizer/-/build-optimizer-0.0.42.tgz",
-      "integrity": "sha512-BAYCVZ10ro6mgZQDZiNiVbX8ppygw4q7z/stpwG8WjMswgMRIcxsxYoC1VFuWcUPAf4UyfTIav6e8UZWA5+xnQ==",
```
Comment (lharzenetter) [inline org.eclipse.winery.topologymodeler.ui/package-lock.json]: Please ignore these changes. We require the sha512 hash.

File: org.eclipse.winery.topologymodeler.ui/src/app/node/node.component.ts [hunk 2]
```diff
@@ -143,44 +142,44 @@ export class NodeComponent implements OnInit, AfterViewInit, OnDestroy, DoCheck
      * We have 3 possibilities: none, XML element, or Key value pairs.
      * @param {string} type
      */
-    findOutPropertyDefinitionTypeForProperties(type: string, groupedNodeTypes: any): string {
-        for (const nameSpace of groupedNodeTypes) {
-            for (const nodeTypeVar of nameSpace.children) {
+    findOutPropertyDefinitionTypeForProperties(type: string, groupedNodeTypes: any): void {
```
Comment (lharzenetter) [inline org.eclipse.winery.topologymodeler.ui/src/app/node/node.component.ts]: Why is `groupedNodeTypes` of type `any`?

Comment (lharzenetter) [reply org.eclipse.winery.topologymodeler.ui/src/app/node/node.component.ts]: Oh I already saw these changes in https://github.com/eclipse/winery/pull/249. What did you do here?

File: org.eclipse.winery.topologymodeler.ui/src/app/services/error-handler.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { TestBed, inject } from '@angular/core/testing';
```
Comment (lharzenetter) [inline org.eclipse.winery.topologymodeler.ui/src/app/services/error-handler.service.spec.ts]: We don't have tests yet, so please delete the auto generated ones.

File: org.eclipse.winery.topologymodeler.ui/src/app/services/backend.service.ts [hunk 4]
```diff
@@ -469,6 +479,15 @@ export class BackendService {
         return this.http.get(url + '/', { headers: this.headers });
     }
 
+    /**
+     * Requests all topology template ids
+     * @returns {Observable<string>}
+     */
+    requestAllTopologyTemplates(): Observable<any> {
```
Comment (lharzenetter) [inline org.eclipse.winery.topologymodeler.ui/src/app/services/backend.service.ts]: Why do you return `any` here? Don't you know the type you get?

File: org.eclipse.winery.topologymodeler.ui/package-lock.json (hunk 1 above)
Comment (zelenikty) [reply org.eclipse.winery.topologymodeler.ui/package-lock.json]: done

File: org.eclipse.winery.topologymodeler.ui/src/app/node/node.component.ts (hunk 2 above)
Comment (zelenikty) [reply org.eclipse.winery.topologymodeler.ui/src/app/node/node.component.ts]: done

File: org.eclipse.winery.topologymodeler.ui/src/app/services/error-handler.service.spec.ts (hunk 3 above)
Comment (zelenikty) [reply org.eclipse.winery.topologymodeler.ui/src/app/services/error-handler.service.spec.ts]: done

File: org.eclipse.winery.topologymodeler.ui/src/app/services/backend.service.ts (hunk 4 above)
Comment (zelenikty) [reply org.eclipse.winery.topologymodeler.ui/src/app/services/backend.service.ts]: corrected

**Pull request merged**